pkg/northbound/admin/admin.go

```diff
@@ -175,3 +177,62 @@ func (s Server) RollbackTransaction(ctx context.Context, req *admin.RollbackRequ
 	}
 	return nil, ctx.Err()
 }
+
+// LeafSelectionQuery selects values allowable for leaf.
+func (s Server) LeafSelectionQuery(ctx context.Context, req *admin.LeafSelectionQueryRequest) (*admin.LeafSelectionQueryResponse, error) {
+	log.Debugf("Received LeafSelectionQuery %+v", req)
+	logContext(ctx, "LeafSelectionQuery()")
+	if req == nil {
+		return nil, errors.NewInvalid("request is empty")
+	}
+
+	groups := make([]string, 0)
+	if md := metautils.ExtractIncoming(ctx); md != nil && md.Get("name") != "" {
+		groups = append(groups, strings.Split(md.Get("groups"), ";")...)
+		log.Debugf("gNMI LeafSelectionQuery() called by '%s (%s)'. Groups %v. Token %s",
+			md.Get("name"), md.Get("email"), groups, md.Get("at_hash"))
+	}
+
+	configType := configapi.TargetType(req.Type)
+	configVersion := configapi.TargetVersion(req.Version)
+
+	config, err := s.configurationsStore.Get(ctx,
+		configuration.NewID(configapi.TargetID(req.Target), configType, configVersion))
+	if err != nil {
+		return nil, errors.NewInvalid("error retrieving config for %s %s %s %v",
+			req.Target, configType, configVersion, err)
+	}
+
+	if req.ChangeContext != nil &&
+		len(req.ChangeContext.GetUpdate())+len(req.ChangeContext.GetReplace())+len(req.ChangeContext.GetDelete()) > 0 {
+
+		log.Warnf("Ignoring change context for the moment")
```

Comment:
Here you can just use warn since you are not printing any other objects

Reply:
In gRPC server all of the errors returned back to the user should uses errors.Status.

Reply:
done